ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/BaseOptimizeCommit.java

```diff
@@ -232,20 +233,27 @@ public long commit(TableOptimizeRuntime tableOptimizeRuntime) throws Exception {
         if (!addDeleteFiles.isEmpty()) {
           throw new IllegalArgumentException("for major optimize, can't add delete files " + addDeleteFiles);
         }
-        if (deleteDeleteFiles.isEmpty()) {
-          OverwriteFiles overwriteFiles = baseArcticTable.newOverwrite();
-          overwriteFiles.set(SnapshotSummary.SNAPSHOT_PRODUCER, CommitMetaProducer.OPTIMIZE.name());
-          deleteDataFiles.forEach(overwriteFiles::deleteFile);
-          addDataFiles.forEach(overwriteFiles::addFile);
-          overwriteFiles.commit();
-        } else {
+
+        // overwrite DataFiles
+        OverwriteFiles overwriteFiles = baseArcticTable.newOverwrite();
+        overwriteFiles.set(SnapshotSummary.SNAPSHOT_PRODUCER, CommitMetaProducer.OPTIMIZE.name());
+        deleteDataFiles.forEach(overwriteFiles::deleteFile);
+        addDataFiles.forEach(overwriteFiles::addFile);
+        overwriteFiles.commit();
```

Comment:
> What if both deleteDataFiles and addDataFiles are empty? If an empty snapshot is produced, this issue #231 must be fixed

For Major commit, it is not impossible both deleteDataFiles and addDataFiles are empty.Because Major optimize need 1 data file at latest